packages/medusa/src/strategies/product-export.ts

```diff
@@ -0,0 +1,781 @@
+        continue
+      }
+      productLineData.push("")
+    }
+
+    productLineData.push(this.NEWLINE)
+    return productLineData.join(this.DELIMITER_)
+  }
+
+  private buildProductVariantsLineData(
+    product: Product,
+    variants: ProductVariant[] = []
+  ): string[] {
+    const outputLineData: string[] = []
+
+    for (const variant of variants) {
+      const variantLineData: string[] = []
+
+      for (const [
+        columnTitle,
+        columnSchema,
+      ] of this.columnDescriptors.entries()) {
+        if (
+          columnTitle === "Product Handle" &&
+          columnSchema.entityName === "product"
+        ) {
+          variantLineData.push(columnSchema.accessor(product))
+          continue
+        }
+
+        if (columnSchema.entityName === "variant") {
+          variantLineData.push(columnSchema.accessor(variant))
+          continue
+        }
+        if (
+          columnSchema.entityName === "variantOption" &&
+          columnSchema.index !== undefined
+        ) {
+          const option = variant?.options[columnSchema.index] ?? {}
+          variantLineData.push(columnSchema.accessor(option))
+          continue
+        }
+        if (
+          columnSchema.entityName === "moneyAmount" &&
+          columnSchema.index !== undefined
+        ) {
+          const moneyAmount = variant?.prices[columnSchema.index] ?? {}
+          variantLineData.push(columnSchema.accessor(moneyAmount))
+          continue
+        }
+        variantLineData.push("")
+      }
+
+      variantLineData.push(this.NEWLINE)
+      outputLineData.push(variantLineData.join(this.DELIMITER_))
+    }
+
+    return outputLineData
+  }
+
+  private async handleProcessingErrors(
+    batchJobId: string,
+    err: unknown,
+    {
+      offset,
+      count,
+      advancementCount,
+      progress,
+    }: {
+      offset: number
+      count: number
+      advancementCount: number
+      progress: number
+    }
+  ): Promise<void> {
+    return await this.atomicPhase_(async (transactionManager) => {
+      const batchJob = await this.batchJobService_
+        .withTransaction(transactionManager)
+        .retrieve(batchJobId)
+
+      const retryCount = batchJob.context.retry_count ?? 0
+      const maxRetry = batchJob.context.max_retry ?? 0
+
+      if (err instanceof MedusaError) {
+        await this.batchJobService_
+          .withTransaction(transactionManager)
+          .updateStatus(batchJob, BatchJobStatus.FAILED)
+      } else if (retryCount < maxRetry) {
+        await this.batchJobService_
+          .withTransaction(transactionManager)
+          .update(batchJobId, {
+            context: {
+              ...batchJob.context,
+              retry_count: retryCount + 1,
+              offset,
+            },
+            result: {
+              ...batchJob.result,
+              count,
+              advancementCount,
+              progress,
+              err,
+            },
+          })
```

Comment:
maybe we should go back to the confirmed state? so that the worker will automatically pick it up again?

Reply:
maybe, we could also just create a "retry" method which emits the correct event (preprocessing or processing)?

Reply:
but we don't want it to be consume priority to something else no? sending an event would make the worker to consume it immediatly event if something else was waiting? putting it back to the previous status would allow to flow to continue as usual wdyt?